src/model.py

```diff
@@ -0,0 +1,145 @@
+import helpers
+import classifiers
+import pandas as pd
+import matplotlib.pyplot as plt
+from sklearn.metrics import classification_report
+from sklearn.metrics import confusion_matrix
+from sklearn.decomposition import RandomizedPCA
+from config import *
+import os.path
+import pickle
+
+# Interface between the server/local helper and the neural network
+class Model(object):
+    def __init__(self, preprocessor, clf_id=MLP_ID, db=None):
+        self.preprocessor = preprocessor
+        self.clf_id = clf_id
+        self.clf = classifiers.choose_clf(clf_id)
+        self.db = db
+
+    def initialize(self):
+        cache_path = self.clf.cache_string()
+        if LOAD_CACHE and os.path.isfile(cache_path):
+            print("Loading model from %s" % cache_path)
+            with open(cache_path, 'rb') as f:
+                self.clf = pickle.load(f)
+        else:
+            print("Initializing model")
+            if self.db:
+                print("Loading data from DB")
+                data = self.preprocessor.data_from_db()
+            else:
+                print("Loading data from sklearn")
+                data = self.preprocessor.get_data()
+
+            (X_train, 
+            X_test, \
+            y_train, \
+            y_test, \
+            target_names, \
+            H, \
+            W
+            ) = data
+
+            pca = RandomizedPCA(n_components=PCA_N_COMPONENTS, whiten=True).fit(X_train)
```

Comment:
No, what this does is tries to reduce feature dimensionality. It tries to cover the variance with fewer dimensions and keeps only the more relevant/impactful ones. It's actually a preprocessing step but I'm not sure why I get an error when I try to put this work into a function and return a pca instance. The whitening does some type of contrasting to try to make things more linearly separable or so we can notice feature effects more easily.
https://en.wikipedia.org/wiki/Principal_component_analysis